atom/index.js

```diff
@@ -1,64 +1,49 @@
-import { clean } from '../clean-stores/index.js'
-
-export const STORE_CLEAN_DELAY = 1000
-
-export function atom(init) {
+export const atom = value => {
   let currentListeners
   let nextListeners = []
-  let destroy
-
   let store = {
-    set(newValue) {
-      store.value = newValue
+    lc: 0,
+    value,
+    set(data) {
+      store.value = data
+      store.notify()
+    },
+    get() {
+      let unsub
+      if (!store.lc) unsub = store.listen(() => {})
+      unsub && unsub()
+      return store.value
+    },
+    notify(changedKey) {
       currentListeners = nextListeners
       for (let listener of currentListeners) {
-        listener(store.value)
+        listener(store.value, changedKey)
       }
     },
-
-    subscribe(listener) {
-      let unbind = store.listen(listener)
-      listener(store.value)
-      return unbind
-    },
-
     listen(listener) {
-      if (!store.active) {
-        store.active = true
-        if (init) destroy = init()
-      }
       if (nextListeners === currentListeners) {
         nextListeners = nextListeners.slice()
       }
-      nextListeners.push(listener)
+      store.lc = nextListeners.push(listener)
       return () => {
         if (nextListeners === currentListeners) {
           nextListeners = nextListeners.slice()
         }
         let index = nextListeners.indexOf(listener)
         nextListeners.splice(index, 1)
-        if (!nextListeners.length) {
-          setTimeout(() => {
-            if (store.active && !nextListeners.length) {
-              if (destroy) destroy()
-              destroy = undefined
-              store.active = undefined
-            }
-          }, STORE_CLEAN_DELAY)
-        }
+        store.lc--
```

Comment:
Hey 👋 Looks like this might produce incorrect results if unsubscribe function is called multiple times

Reply:
Oh 😓 Can you show some example?

Reply:
For example `atom/index.test.ts`

```typescript
it('listen', () => {
  expect.assertions(4)
  let store = atom({ some: { path: 0 } })
  let unbind = store.listen(value => {
    expect(value).toBe(store.get())
  })

  store.set({ some: { path: 1 } })
  store.set({ some: { path: 2 } })
  expect(store.get()).toEqual({ some: { path: 2 } })
  unbind()
  unbind()
  expect(store.lc).toEqual(-1)
})
```
would pass, meaning `store.lc` checks might not work as expected. I am not sure how bad this can be. Just something I stumbled upon when trying to extend Svelte's stores with some extra optimisations.

Reply:
In my case subscribers are kept in a Set, so I just check its size instead of keeping a separate counter

Reply:
Thanks. Fixed cc4d514]

I tried `Set` for listeners, but it leads to drop of 25% [benchmark](https://github.com/nanostores/nanostores/blob/next/test/benchmark.js) results